### ISSUE TITLE
TS project that depends on gcs-resumable-upload doesn't compile

### ISSUE THREAD

**Issue by kinwa91** [stephenplusplus/gcs-resumable-upload#69]
I am converting `@google-cloud/storage` to typescript and I got this compilation error from gcs-resumable-upload version 0.11.0:

![screen shot 2018-06-22 at 2 44 21 pm](https://user-images.githubusercontent.com/4001432/41800561-e6e6acaa-762a-11e8-83a9-2c161e8372b6.png)

Compiling from 51c7231 onwards produce working typescript declarations.

**Comment by kinwa91** [comment]
@stephenplusplus can we create a release on the latest master?

**Comment by kinwa91** [comment]
it was the update to `typescript: ~2.9.0` via 51c7231

**Comment by kinwa91** [comment]
from 2.9 onwards, npm run compile generates:
```typescript
/// <reference types="node" />
import { AxiosRequestConfig, AxiosResponse } from 'axios';
import * as ConfigStore from 'configstore';
import { GoogleAuth, GoogleAuthOptions } from 'google-auth-library';
import * as Pumpify from 'pumpify';
import * as r from 'request';
export declare type RequestBody = any;
export declare type RequestResponse = AxiosResponse;
export declare type Request = r.Request;
export declare type RequestOptions = AxiosRequestConfig;
export declare type RequestCallback = (err: Error | null, response?: AxiosResponse, body?: RequestBody) => void;
export declare type AuthorizeRequestCallback = (err: Error | null, authorizedReqOpts: RequestOptions) => void;
export declare type CreateUriCallback = (err: Error | null, uri?: string) => void;
export interface Encryption {
    key: {};
    hash: {};
}
export interface UploadConfig {
    /**
     * The name of the destination bucket.
     */
    bucket: string;
    /**
     * The name of the destination file.

...
```

prior to commit 51c7231:

```typescript
/// <reference types="request" />
/// <reference types="node" />
/// <reference types="pumpify" />
/// <reference types="configstore" />
import { AxiosRequestConfig, AxiosResponse } from 'axios';
import * as ConfigStore from 'configstore';
import { GoogleAuth, GoogleAuthOptions } from 'google-auth-library';
import * as Pumpify from 'pumpify';
import * as r from 'request';
export declare type RequestBody = any;
export declare type RequestResponse = AxiosResponse;
export declare type Request = r.Request;
export declare type RequestOptions = AxiosRequestConfig;
export declare type RequestCallback = (err: Error | null, response?: AxiosResponse, body?: RequestBody) => void;
export declare type AuthorizeRequestCallback = (err: Error | null, authorizedReqOpts: RequestOptions) => void;
export declare type CreateUriCallback = (err: Error | null, uri?: string) => void;
export interface Encryption {
    key: {};
    hash: {};
}
export interface UploadConfig {
    /**

...
```

**Comment by stephenplusplus** [comment]
Thank you! Published 0.11.1 👍

**Comment by kinwa91** [comment]
good stuff. Thanks! @stephenplusplus
